### PR TITLE
Add ac001 keyboard!

### PR DESCRIPTION
**Pull request by jonylee1986**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jonylee1986** [comment]
Add m1 and mk47 keyboard,TKS!

**Comment by jonylee1986** [comment]
Add m1 and mk47 keyboard,TKS!

File: keyboards/hfdkb/ac001/ac001.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
+	K000, K001, K002,  K003, K004  \
+) { \
+	{ K000,  K001,  K002,  KC_NO,  KC_NO } \
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/ac001.h:24]:
```suggestion
	{ K000,  K001,  K002,  K003,  K004 } \
```
Or is there a good reason for this?

File: keyboards/hfdkb/ac001/config.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#define BOOTMAGIC_LITE_COLUMN 0
+
+#define TAP_CODE_DELAY 5
+
+/* ws2812 RGB LED */
+#define RGB_DI_PIN A1
+#define DRIVER_LED_TOTAL    5
+#define RGBLED_NUM          5        // Number of LEDs
+#define NOP_FUDGE           0.4
+
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/config.h:47]:
```suggestion
 #define FORCE_NKRO

/* key matrix pins */
#define MATRIX_ROW_PINS { B15}
#define MATRIX_COL_PINS { A5, A6, A7, C4, C5}

/* COL2ROW or ROW2COL */
#define DIODE_DIRECTION ROW2COL

#define TAP_CODE_DELAY 5

/* ws2812 RGB LED */
#define RGB_DI_PIN A1
#define RGB_MATRIX_LED_COUNT 5
```

File: keyboards/hfdkb/ac001/config.h [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+/* Send up to 4 key press events per scan */
+#define QMK_KEYS_PER_SCAN 4
+
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_LIMIT_VAL 180 /* The maximum brightness level */
+
+/* Set debounce time to 1ms */
+#define DEBOUNCE 5
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/config.h:130]:
```suggestion
```

File: keyboards/hfdkb/ac001/keymaps/via/config.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/keymaps/via/config.h:20]:
```suggestion
```

File: keyboards/hfdkb/ac001/keymaps/via/config.h [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#define STM32_USB_USE_OTG1 TRUE
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/keymaps/via/config.h:21]: Why is this specific to VIA keymap? Does WB32 even support USB OTG (in ChibiOS)? If so, pretty sure it should be `WB32_...`.

File: keyboards/hfdkb/ac001/rules.mk [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = WB32FQ95
+
+# Bootloader selection
+BOOTLOADER = wb32-dfu
+
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/rules.mk:6]:
```suggestion
```

File: keyboards/hfdkb/ac001/info.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
+    "keyboard_name": "ac001",
+    "manufacturer": "hfd",
+    "url": "",
+    "maintainer": "jonylee@hfd",
+    "usb": {
+        "vid": "0xFFFE",
+        "pid": "0x0001",
+        "device_version": "0.0.1"
+    },
```
Comment (waffle87) [inline keyboards/hfdkb/ac001/info.json:10]:
```suggestion
    },
    "processor": "WB32FQ95",
    "bootloader": "wb32-dfu",
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by jonylee1986** [comment]
Please review again. @waffle87

**Review by waffle87 — APPROVED** [review]

**Comment by zvecr** [comment]
Your recent git actions have pulled in a bunch of crap and will make this PR unmergable.

**Comment by jonylee1986** [comment]
I'm sorry, but this problem is caused by my negligence. I will sort out the code of ac001 and open a new PR @zvecr

**Pull request closed without merging**